### PR TITLE
Allow creation of new (virtual) section from .pages

### LINKED ISSUE #37: Grouping of Rest items is not working
I'm experiencing a `TypeError` error with the following configuration:

**.pages:**
```yaml
order: desc
nav:
  - 2020 Blogs:
    - ... | regex=2020[A-Z0-9a-z_-]+.md
  - 2019 Blogs:
    - ... | regex=2019[A-Z0-9a-z_-]+.md
```

However the following configuration is working:
```yaml
order: desc
nav:
  - ... | regex=2020[A-Z0-9a-z_-]+.md
```

**Error:**
```python
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/__main__.py", line 133, in serve_command
    serve.serve(
  File "/usr/local/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 141, in serve
    config = builder()
  File "/usr/local/lib/python3.8/site-packages/mkdocs/commands/serve.py", line 136, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/commands/build.py", line 266, in build
    nav = config['plugins'].run_event('nav', nav, config=config, files=files)
  File "/usr/local/lib/python3.8/site-packages/mkdocs/plugins.py", line 94, in run_event
    result = method(item, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/mkdocs_awesome_pages_plugin/plugin.py", line 45, in on_nav
    return AwesomeNavigation(nav.items, Options(**self.config), config['docs_dir'], explicit_sections).to_mkdocs()
  File "/usr/local/lib/python3.8/site-packages/mkdocs_awesome_pages_plugin/navigation.py", line 43, in __init__
    self.meta = NavigationMeta(items, options, docs_dir, explicit_sections)
  File "/usr/local/lib/python3.8/site-packages/mkdocs_awesome_pages_plugin/navigation.py", line 183, in __init__
    self.root = Meta.try_load_from(join_paths(root_path, self.options.filename))
  File "/usr/local/lib/python3.8/site-packages/mkdocs_awesome_pages_plugin/meta.py", line 131, in try_load_from
    return Meta.load_from(path)
  File "/usr/local/lib/python3.8/site-packages/mkdocs_awesome_pages_plugin/meta.py", line 175, in load_from
    nav = [MetaNavItem.from_yaml(item, path) for item in nav]
  File "/usr/local/lib/python3.8/site-packages/mkdocs_awesome_pages_plugin/meta.py", line 175, in <listcomp>
    nav = [MetaNavItem.from_yaml(item, path) for item in nav]
  File "/usr/local/lib/python3.8/site-packages/mkdocs_awesome_pages_plugin/meta.py", line 41, in from_yaml
    raise TypeError('Invalid nav item format {type} [{context}]'.format(type=item, context=context))
TypeError: Invalid nav item format {'2020 Blogs': ['... | regex=2020[A-Z0-9a-z_-]+.md']} [/docs/docs/.pages]
```

Also I'm building a custom Docker image for [Material for MkDocs](https://squidfunk.github.io/mkdocs-material/), to include the **awesome-pages-plugin**: - https://squidfunk.github.io/mkdocs-material/getting-started/ - `How to add plugins to the Docker image?`

### PR DESCRIPTION
**Pull request by dimrozakis**
Allows the `nav` config of `.pages` to create new sections to organize
pages.

Fixes #37

### CODE REVIEW

**Comment by dimrozakis** [comment]
Hi @lukasgeiter 

I created this PR because I wanted to reorganize the pages in a more nested hierarchy without changing their paths (and their links). This is working properly for me in the site I'm managing. All preexisting tests pass and I've added a couple of new tests for the new functionality.

Some notes:

1. I'm not very familiar with `mkdocs-awesome-pages-plugin`. This was a quick implementation which I'm sure could be improved upon. I'm not sure if you're happy with the use of nested functions. Or the term "virtual" section. Feel free to make suggestions or direct edits.
2. As previously said, this is working for me and all preexisting and new tests pass, but I haven't covered every possible use case in the tests. For example I don't test nested "virtual" sections. Or what happens if a virtual section has the same name as a real section. I can't 100% guarantee that this works properly in combination with every other feature of the plugin.
3. I haven't documented this feature in the README.
4. I'm not sure but I think that this can also satisfy the use case of https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin#combine-custom-navigation--file-structure with a top level .pages instead of defining the nav section in mkdocs.yml.

Looking forward to your feedback. And thanks for creating this plugin.

**Comment by lukasgeiter** [comment]
Hey @dimrozakis 

Thanks for the PR! Right now I'm a bit short on time, but I'll take a closer look in the coming days.

File: mkdocs_awesome_pages_plugin/meta.py [hunk 1]
```diff
@@ -34,8 +34,11 @@ def from_yaml(item: Union[str, dict], context: str):
 
         if isinstance(item, dict) and len(item) == 1:
             (title, value) = list(item.items())[0]
-            if isinstance(value, str) and isinstance(title, str):
-                return MetaNavItem(value, title)
+            if isinstance(title, str):
+                if isinstance(value, str):
+                    return MetaNavItem(value, title)
+                elif isinstance(value, list):
+                    return MetaNavItem([MetaNavItem.from_yaml(it, context) for it in value], title)
```
Comment (lukasgeiter) [inline mkdocs_awesome_pages_plugin/meta.py:41]: The type annotation of `value` needs to be extended for this. `Union[str, List[MetaNavItem]]` should do the trick.

File: mkdocs_awesome_pages_plugin/navigation.py [hunk 2]
```diff
@@ -37,6 +37,10 @@ def __init__(self, filename: str):
         )
 
 
+class VSection(Section):
```
Comment (lukasgeiter) [inline mkdocs_awesome_pages_plugin/navigation.py:40]: Could you spell this out as `VirtualSection`? Please also do so for any other instances of the term.

File: mkdocs_awesome_pages_plugin/navigation.py [hunk 3]
```diff
@@ -86,19 +90,25 @@ def _nav(self, items: List[NavigationItem], meta: Meta) -> List[NavigationItem]:
...
-            else:
-                if meta_item.value in items_by_basename:
+        def _make_nav_rec(meta_nav):
+            result = []
+            for meta_item in meta_nav:
+                if isinstance(meta_item, MetaNavRestItem):
+                    rest_items.append(meta_item)
+                    result.append(meta_item)
+
+                elif isinstance(meta_item.value, (list, tuple)):
```
Comment (lukasgeiter) [inline mkdocs_awesome_pages_plugin/navigation.py:103]: Is `tuple` needed here?

File: mkdocs_awesome_pages_plugin/navigation.py [hunk 4]
```diff
@@ -86,19 +90,25 @@ def _nav(self, items: List[NavigationItem], meta: Meta) -> List[NavigationItem]:
...
+        used_items = []
         rest_items = RestItemList()
 
-        for index, meta_item in enumerate(meta.nav):
-            if isinstance(meta_item, MetaNavRestItem):
-                rest_items.append(meta_item)
-                result.append(meta_item)
-            else:
-                if meta_item.value in items_by_basename:
+        def _make_nav_rec(meta_nav):
```
Comment (lukasgeiter) [inline mkdocs_awesome_pages_plugin/navigation.py:96]: Please add a type annotation to the parameter here

File: mkdocs_awesome_pages_plugin/navigation.py [hunk 5]
```diff
@@ -109,21 +119,30 @@ def _nav(self, items: List[NavigationItem], meta: Meta) -> List[NavigationItem]:
...
                     path = basename(self._get_item_path(item))
                     for rest_item in rest_items:
                         if rest_item.matches(path):
                             rest[rest_item].append(item)
                             break
 
-            for index, item in enumerate(result):
-                if isinstance(item, MetaNavRestItem):
-                    result[index : index + 1] = rest[item]
+            def _expand_rest_rec(result):
```
Comment (lukasgeiter) [inline mkdocs_awesome_pages_plugin/navigation.py:137]: Please add a type annotation to the parameter here

**Review by lukasgeiter — CHANGES_REQUESTED** [review]
Hey there, sorry for the long wait! There are few stylistic changes I'd like you to make, but otherwise it's looking pretty good 👍

**Comment by dimrozakis** [comment]
@lukasgeiter thanks for the review. I've addressed all your comments. Any thoughts on points 2,3,4 from https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin/pull/59#issuecomment-1021559894?

**Review by lukasgeiter — APPROVED** [review]
Great, thanks for the changes! My thoughts on your previous comment:

2. I think it's fine. I do appreciate your concern though!
3. No worries, I'll take care of that.
4. Almost, but not quite. There's still a difference in which files can be targeted. A `.pages` can only access the files and folders in the same directory, while  `mkdocs.yml` has access to everything.

**Pull request merged**